django_eth_events/decoder.py

```diff
@@ -125,21 +125,39 @@ def decode_log(self, log):
         decoded_event = {
             'params': decoded_params,
             'name': method['name'],
-            'address': self.decode_address(log['address'])
+            'address': self.decode_address(log['address']),
+            'transaction_hash': self.decode_transaction(log['transactionHash'])
         }
 
         return decoded_event
 
     @staticmethod
     def decode_address(address):
+        if not address:
+            raise ValueError
+
         if isinstance(address, bytes):
             address = address.hex()
+
         # Address length must be 40 (42 with 0x), but usually it's packed on 32 bits (length of 66 with 0x)
         if len(address) == 66:
             address = address[26:]
 
         return normalize_address_without_0x(address)
 
+    @staticmethod
+    def decode_transaction(tx_hash):
+        if not tx_hash:
+            raise ValueError
+
+        if isinstance(tx_hash, (bytes, HexBytes)):
+            tx_hash = tx_hash.hex()
```

Comment:
Be careful here. `hex()` function is different for `bytes` and for `HexBytes`. `HexBytes` always uses `0x` prefix, `bytes` doesn't, e.g:

```
In [3]: HexBytes('0x05').hex()                                      
Out[3]: '0x05'

In [4]: bytes(HexBytes('0x05')).hex()                               
Out[4]: '05'
```

Reply:
good catch, thanks 👍

Reply:
should be resolved now